lib/shared/sdk/usbboot/index.js

```diff
@@ -251,14 +261,31 @@ class USBBootAdapter extends EventEmitter {
     usb.listDevices().filter(isUsbBootCapableUSBDevice).map((device) => {
     /* eslint-enable lodash/prefer-lodash-method */
 
+      const description = COMPUTE_MODULE_DESCRIPTIONS[device.deviceDescriptor.idProduct] || 'Compute Module'
+
+      if (!device.accessible) {
+        return {
+          device: `${usbIdToString(device.deviceDescriptor.idVendor)}:${usbIdToString(device.deviceDescriptor.idProduct)}`,
+          displayName: 'Missing drivers',
+          description,
+          mountpoints: [],
+          protected: false,
+          system: false,
+          disabled: true,
+          icon: 'warning',
+          size: null,
+          adaptor: USBBootAdapter.id
+        }
+      }
+
       // This is the only way we can unique identify devices
       device.raw = `${device.busNumber}:${device.deviceAddress}`
 
       const result = {
         device: device.raw,
         raw: device.raw,
         displayName: 'Initializing device',
-        description: 'Compute Module',
+        description,
```

Comment:
Is this `description` here only shown before `bootcode.bin` has been written to the Pi, or also afterwards? If it's the latter, remember that after `bootcode.bin` has been written, both the CM1 and CM3 re-enumarate as the identical VID:PID combo of 0x0a5c:0x2764 which is why this previously said only "Compute Module" rather than being more specific.

Whereas in the `if (!device.accessible) {` case, we know `bootcode.bin` definitely hasn't been written to the device yet, so at that point it _is_ safe to differentiate by PID.

Reply:
Ah, true, let me fix that.

Reply:
Done, glad you caught this :D

Reply:
Of course, once we add the USB-hardware-path stuff into Etcher (needed for Etcher Pro) then we'll be able to change all this yet again as we'll be able to uniquely track devices by which physical slot they're plugged into, so we'll then know what a device initially enumerated as, even after it re-enumerates with 0x0a5c:0x2764 :-)